src/app/[username]/page.tsx

```diff
@@ -60,52 +62,74 @@ const ProfilePage = ({ params }: { params: { username: string } }) => {
   const disableHireMe = params.username === user?.accountId.split('.')[0];
   const freelancerFullName = `${freelancer?.firstName} ${freelancer?.lastName}`;
 
+  const handleClaimMyProfile = () => {
+    if (user) {
+      logout();
+    }
+    router.push('/signin');
+  };
+
   return (
     <>
-      <Header />
-      {!isPending && freelancer && (
+      {isError ? (
+        <ClaimYourProfile
+          username={params.username}
+          handleClaimMyProfile={handleClaimMyProfile}
+        />
+      ) : (
         <>
-          <Profile
-            user={freelancer}
-            canEdit={false}
-            onHireMeClick={() => setOfferDialogOpen(true)}
-            disableHireMe={disableHireMe}
-          />
-          <UserSocialPreview title={freelancerFullName} />
-          <ProjectOfferDialog
-            isOpen={isOfferDialogOpen}
-            onClose={(open) => setOfferDialogOpen(open)}
-            onSubmit={mutation.mutate}
-            name={freelancerFullName}
-            formSubmitted={mutation.isSuccess}
-          />
+          <Header />
+          {!isPending && freelancer && (
+            <>
+              <Profile
+                user={freelancer}
+                canEdit={false}
+                onHireMeClick={() => setOfferDialogOpen(true)}
+                disableHireMe={disableHireMe}
+              />
+              <UserSocialPreview title={freelancerFullName} />
+              <ProjectOfferDialog
+                isOpen={isOfferDialogOpen}
+                onClose={(open) => setOfferDialogOpen(open)}
+                onSubmit={mutation.mutate}
+                name={freelancerFullName}
+                formSubmitted={mutation.isSuccess}
+              />
+            </>
+          )}
         </>
       )}
-      {isError && <ClaimYourProfile username={params.username} />}
     </>
   );
 };
 
 export default ProfilePage;
 
 /** Local component to display a "Claim your profile CTA when visiting a profile that does not exist" */
-const ClaimYourProfile = (props: { username: string }) => {
+const ClaimYourProfile = (props: {
+  username: string;
```

Comment:
```suggestion
  handle: string;
```
I think I want to move in the direction of calling this a handle. What do you think?